src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -2628,10 +2628,10 @@ private void ParseExpectedErrorConfigurations()
             var expectedStatusCodesArrayLocal = _localConfiguration.errorCollector.expectedStatusCodes?.Split(StringSeparators.Comma, StringSplitOptions.RemoveEmptyEntries);
             var expectedStatusCodesArrayServer = _serverConfiguration.RpmConfig.ErrorCollectorExpectedStatusCodes;
 
-            var expectedStatusCodesArray = ServerOverrides(expectedStatusCodesArrayServer, expectedStatusCodesArrayLocal);
+            var expectedStatusCodesArray = EnvironmentOverrides(ServerOverrides(expectedStatusCodesArrayServer, expectedStatusCodesArrayLocal)?.ToList(), "NEW_RELIC_ERROR_COLLECTOR_EXPECTED_ERROR_CODES");
```

Comment:
Is the ToList call necessary?

Reply:
Yes, because of the method signature of the overload of `EnvironmentOverrides` being used: https://github.com/newrelic/newrelic-dotnet-agent/blob/ccde882ae97011eb878698c23163741c66851c39/src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs#L2385

Is it worth exploring having that take and return an `IEnumerable<string>` instead?

Reply:
If expectedStatusCodesArrayLocal, or expectedStatusCodesArrayLocal were already lists, then the ToList call would not be necessary.

Reply:
Actually, it wasn't that hard to do. 👍